### DISCUSSION TITLE
Trouble displaying video using timelineVariable

### DISCUSSION THREAD

**Discussion by latifn2** [jspsych/jsPsych#1041] [Q&A] [answered]
Hi, 

I apologize for the naive question, I'm very new to this. I really appreciate any help. 

I have a fairly simply experiment where I want to display videos (randomized) followed by a question. To do this, I created a procedure to run the videos with the questions. However, my videos are simply not displaying and I cannot figure out why. Further, I want to force participants to watch the video without being able to end it and I want it to automatically go to the question after the video ends. However, because the video never displays, it also never ends and thus the whole procedure gets stuck. There are no errors in the console. Thanks so much in advance for any help. I've pasted my code below: 

```js
/* test trials */

    var test_stimuli = [
      { stimulus: "10_NS2_Right_After_20.mp4"},
      { stimulus: "11_NS2_Right_Before_30.mp4" }
    ];


    var fixation = {
          type: 'html-keyboard-response',
          stimulus: '<div style="font-size:60px;">+</div>',
          choices: jsPsych.NO_KEYS,
          trial_duration: 500,
    }

    var stimuli_list = {
      type: 'video-keyboard-response',
      sources:jsPsych.timelineVariable('stimulus'),
      trial_ends_after_video:true,
      choices: jsPsych.NO_KEYS,
      width: 800

    }

    var response = {
      type: "html-keyboard-response",
      stimulus: '<div style="font-size:60px;">Did the talker finish their turn?</div>',
      choices: ['y', 'n'],

    }

    var test_procedure = {
      timeline: [fixation, stimuli_list, response],
      timeline_variables: test_stimuli,
      randomize_order: true
    }
    timeline.push(test_procedure);
```

**Comment by jodeleeuw** [comment, marked as answer]
Hi @latifn2 ,

I'm not sure if this is necessary off the top of my head, but the `sources` parameter in the video plugins generally expects an array of sources, not a single path. Try changing the definition of your timeline variables to be an array:

```js
var test_stimuli = [
      { stimulus: ["10_NS2_Right_After_20.mp4"]},
      { stimulus: ["11_NS2_Right_Before_30.mp4"] }
];
```

The other thing that comes to mind is to try preloading your videos manually in the `jsPsych.init(...)` function:

```js
jsPsych.init({
	preload_video: ["10_NS2_Right_After_20.mp4", "11_NS2_Right_Before_30.mp4"]
})
```

**Reply by latifn2** [reply]
Making the timeline variables an array did it! Thanks so incredibly much!!!!
